crates/ruff_cli/src/commands/format.rs

```diff
@@ -713,41 +714,41 @@ pub(super) fn warn_incompatible_formatter_settings(
             Rule::MissingTrailingComma,
         ] {
             if setting.linter.rules.enabled(rule) {
-                incompatible_rules.push(rule);
+                incompatible_rules.insert(rule);
             }
         }
+    }
 
-        // Rules asserting for space indentation
-        if setting.formatter.indent_style.is_tab() {
-            for rule in [Rule::TabIndentation, Rule::IndentWithSpaces] {
-                if setting.linter.rules.enabled(rule) {
-                    incompatible_rules.push(rule);
-                }
-            }
-        }
+    if !incompatible_rules.is_empty() {
+        let mut rule_names: Vec<_> = incompatible_rules
+            .into_iter()
+            .map(|rule| format!("`{}`", rule.noqa_code()))
+            .collect();
+        rule_names.sort();
+        warn_user_once!("The following rules may cause conflicts when used with the formatter: {}. To avoid unexpected behavior, we recommend disabling these rules, either by removing them from the `select` or `extend-select` configuration, or adding then to the `ignore` configuration.", rule_names.join(", "));
```

Comment:
Minor question — since `warn_user_once` works using the call site, does this mean if we find a second set of incompatible rules in a different pyproject file we will not emit a second message with the different set? Probably not common... but not ideal?

Reply:
No, for this _specific_ warning, we iterate over all `pyproject.toml` and collect all incompatible rules, so they would all be flagged here.

(All the other warnings are static, this is the only warning with dynamic input.)

Reply:
Sounds good!